app/src/main/java/com/islaharper/dawnofandroid/presentation/common/GoogleButton.kt

```diff
@@ -0,0 +1,122 @@
+package com.islaharper.dawnofandroid.presentation.common
+
+import android.content.res.Configuration.UI_MODE_NIGHT_YES
+import androidx.compose.animation.animateContentSize
+import androidx.compose.animation.core.LinearOutSlowInEasing
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.material.CircularProgressIndicator
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Surface
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.islaharper.dawnofandroid.R
+import com.islaharper.dawnofandroid.ui.theme.Shapes
+
+@Composable
+fun GoogleButton(
+    modifier: Modifier = Modifier,
+    loadingState: Boolean = false,
+    primaryText: String = "Sign in with Google",
```

Comment:
Minor:

1. do `primaryText` and `secondaryText` need to be parameters? They don't seem to change with the caller. 
2. can be string resources too?

Reply:
Refactored, in next commit.